src/tools/rbd/action/Pool.cc

```diff
@@ -52,9 +52,72 @@ int execute_init(const po::variables_map &vm,
   return 0;
 }
 
-Shell::Action action(
+void get_arguments_stats(po::options_description *positional,
+                         po::options_description *options) {
+  at::add_pool_options(positional, options);
+  at::add_namespace_option(options, at::ARGUMENT_MODIFIER_NONE);
+  at::add_format_options(options);
+}
+
+int execute_stats(const po::variables_map &vm,
+                  const std::vector<std::string> &ceph_global_init_args) {
+  size_t arg_index = 0;
+  std::string pool_name = utils::get_pool_name(vm, &arg_index);
+  std::string namespace_name = utils::get_namespace_name(vm, &arg_index);
+
+  at::Format::Formatter formatter;
+  int r = utils::get_formatter(vm, &formatter);
+  if (r < 0) {
+    return r;
+  }
+
+  librados::Rados rados;
+  librados::IoCtx io_ctx;
+  r = utils::init(pool_name, namespace_name, &rados, &io_ctx);
+  if (r < 0) {
+    return r;
+  }
+
+  librbd::RBD rbd;
+  uint32_t image_count;
+  uint32_t trash_image_count;
+  uint32_t snap_count;
+  uint64_t provisioned_bytes;
+  r = rbd.pool_provisioned_image_stats_get(io_ctx, true, &image_count,
+                                           &trash_image_count, &snap_count,
+                                           &provisioned_bytes);
+  if (r < 0) {
+    std::cerr << "rbd: failed to query pool stats: " << cpp_strerror(r)
+              << std::endl;
+    return r;
+  }
+
+  if (formatter) {
+    formatter->open_object_section("stats");
+    formatter->dump_unsigned("image_count", image_count);
+    formatter->dump_unsigned("trash_image_count", trash_image_count);
+    formatter->dump_unsigned("snap_count", snap_count);
+    formatter->dump_unsigned("provisioned_bytes", provisioned_bytes);
+    formatter->close_section();
+    formatter->flush(std::cout);
+  } else {
+    std::cout << "Total Images: " << image_count << std::endl
+              << "Total Trashed Images: " << trash_image_count << std::endl
+              << "Total Snapshots: " << snap_count << std::endl
+              << "Provisioned Size: " << stringify(byte_u_t(provisioned_bytes))
```

Comment:
@dillaman Don't we want to totally separate stats for ordinary and trashed images (not only image count)?

May be you have some use case in mind, but just thinking how I would use it, I don't think I would be much interested in trash provisioned stats when looking for the pool provisioned size. I think I would not mind if provisioned size exceeded the pool size due to trashed images. For the trash I would be interested mostly in "used" stats.